packages/create-sitecore-jss/src/templates/nextjs-sxa/src/components/Title.tsx

```diff
@@ -61,17 +63,23 @@ const ComponentContent = (props: ComponentContentProps) => {
 export const Default = (props: TitleProps): JSX.Element => {
   const datasource = props.fields?.data?.datasource || props.fields?.data?.contextItem;
   const { sitecoreContext } = useSitecoreContext();
+  const isMetadataMode = sitecoreContext?.editMode === 'metadata';
+  const editingField = {
+    ...(isMetadataMode
+      ? { metadata: datasource?.field?.jsonValue?.metadata }
+      : { editable: datasource?.field?.jsonValue?.editable }),
+  };
 
   const text: TextField = {
     value: datasource?.field?.jsonValue?.value,
-    editable: datasource?.field?.jsonValue?.editable,
+    ...editingField,
```

Comment:
Can't we simply set the text field directly to `datasource?.field?.jsonValue`?
e.g.
`const text: TextField = datasource?.field?.jsonValue;`

Could then delete the `isMetadataMode` and `editingField` above.